blueman/bluez/Base.py

```diff
@@ -10,34 +10,20 @@
 
 
 class BaseMeta(GObjectMeta):
-    def __call__(cls, *args: str, **kwargs: str) -> "Base":
-        instances: Dict[str, Dict[str, "Base"]] = cls.__dict__.setdefault("__instances__", {})
+    def __call__(cls, **kwargs: str) -> "Base":
+        if not hasattr(cls, "__instances__"):
+            cls.__instances__: Dict[str, Dict[str, "Base"]] = {}
```

Comment:
Just `Dict[str, "Base"]`. No need to use `cls._interface_name` here at all. :slightly_smiling_face:

Reply:
I am confused, cls.__instances__ still contains the interface name and we really need it.

Reply:
Why? It's always the same for a given class, so `__instances__` will always have a single entry. Am I missing something?

Reply:
How I see it is that a single objectpath can and pretty much always exposes
different interfaces. A device may have org.bluez.Device1 and
org.bluez.Network1 and we have a class for each. Although technically
possible (but this is not how we implemented it) we can not call methods
for one interface on a class created for another.

So even though interface_name was removed as keyword argument we still have
to deal with them.

Or I am missing something obvious 😲😁

Reply:
See https://github.com/blueman-project/blueman/commit/aa90fdbb39d92a54491f61e15b27c7a27cb29ed2 for what is technically possible and what would be needed to drop ``interface_name`` from ``Base.__instances__``. I am happy to do this as it cuts down on the classes significantly :smile_cat:

Reply:
> A device may have org.bluez.Device1 and
> org.bluez.Network1 and we have a class for each.

That's my point. Each of our classes only handles a single interface, so there's no need to store instances by interface in the class property. The change to `Base` in https://github.com/blueman-project/blueman/commit/aa90fdbb39d92a54491f61e15b27c7a27cb29ed2 are exactly what I mean.

Reply:
Ah, the obvious part I was missing is that each class will get it's own ``cls.__instances__``. I thought that regardless of the class BaseMeta would just return the any instance based on the object path. I'll take care of this.